libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxManeuverView.kt

```diff
@@ -58,68 +65,53 @@ class MapboxManeuverView @JvmOverloads constructor(
-                hideSecondaryManeuver()
-            }
-            is ManeuverState.ManeuverSub.Instruction -> {
-                renderSubManeuver(state)
-                renderSubTurnIcon(state)
-            }
-            is ManeuverState.ManeuverSub.Show -> {
-                renderSubManeuverVisibility(VISIBLE)
-            }
-            is ManeuverState.ManeuverSub.Hide -> {
-                renderSubManeuverVisibility(GONE)
-            }
-            is ManeuverState.LaneGuidanceManeuver.AddLanes -> {
-                renderAddLanes(state)
-            }
-            is ManeuverState.LaneGuidanceManeuver.RemoveLanes -> {
-                renderRemoveLanes(state)
-            }
-            is ManeuverState.LaneGuidanceManeuver.Show -> {
-                renderLaneGuidanceVisibility(VISIBLE)
-            }
-            is ManeuverState.LaneGuidanceManeuver.Hide -> {
-                renderLaneGuidanceVisibility(GONE)
-            }
-            is ManeuverState.DistanceRemainingToFinishStep -> {
-                renderDistanceRemaining(state)
+    fun renderManeuver(maneuver: Expected<Maneuver, ManeuverError>) {
+        when (maneuver) {
+            is Expected.Success -> {
+                val primary = maneuver.value.primary
+                val secondary = maneuver.value.secondary
+                val sub = maneuver.value.sub
+                val lane = maneuver.value.laneGuidance
+                if (secondary?.componentList != null) {
+                    updateSecondaryManeuverVisibility(VISIBLE)
+                    renderSecondaryManeuver(secondary)
+                } else {
+                    updateSecondaryManeuverVisibility(GONE)
+                }
+                renderPrimaryManeuver(primary)
+                if (sub?.componentList != null || lane != null) {
+                    updateSubManeuverViewVisibility(VISIBLE)
+                } else {
+                    updateSubManeuverViewVisibility(GONE)
+                }
+                if (sub?.componentList != null) {
+                    renderSubManeuver(sub)
+                } else {
+                    renderSubManeuver(null)
+                }
+                when (lane != null) {
+                    true -> {
+                        renderAddLanes(lane)
+                    }
+                    else -> {
+                        renderRemoveLanes()
+                    }
+                }
             }
-            is ManeuverState.UpcomingManeuvers.Upcoming -> {
-                renderUpcomingManeuvers(state)
-            }
-            is ManeuverState.UpcomingManeuvers.Show -> {
-                renderUpcomingManeuverVisibility(VISIBLE)
-            }
-            is ManeuverState.UpcomingManeuvers.Hide -> {
-                renderUpcomingManeuverVisibility(GONE)
-            }
-            is ManeuverState.UpcomingManeuvers.RemoveUpcoming -> {
-                removeUpcomingManeuver(state)
+            else -> {
```

Comment:
I would not put the `check` inside the view and leave it `empty` because with `check` it will throw `IllegalStateException` and the app will crash. We may want to handle errors differently in the view.

Reply:
Yeah that makes sense. We should come up with a solution on what we want to do in case of failure - topic for a future discussion though as currently we're not generating any 😅